Loop/Managers/SupportManager.swift

```diff
@@ -0,0 +1,258 @@
+//
+//  SupportManager.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 9/8/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import Combine
+import Foundation
+import LoopKit
+import LoopKitUI
+import SwiftUI
+import MockKitUI
+
+public final class SupportManager {
+    
+    private lazy var log = DiagnosticLog(category: "SupportManager")
+
+    private var supports = Locked<[String: SupportUI]>([:])
+    
+    private var identifierWithHighestVersionUpdate: String? {
+        get {
+            return UserDefaults.appGroup?.identifierWithHighestVersionUpdate
+        }
+        set {
+            UserDefaults.appGroup?.identifierWithHighestVersionUpdate = newValue
+        }
+    }
+    
+    private let alertIssuer: AlertIssuer
+    private let pluginManager: PluginManager?
+    private let staticSupportTypes: [SupportUI.Type]
+    private let staticSupportTypesByIdentifier: [String: SupportUI.Type]
+
+    lazy private var cancellables = Set<AnyCancellable>()
+
+    init(pluginManager: PluginManager? = nil,
+         deviceDataManager: DeviceDataManager? = nil,
+         servicesManager: ServicesManager? = nil,
+         staticSupportTypes: [SupportUI.Type]? = nil,
+         alertIssuer: AlertIssuer) {
+        
+        self.alertIssuer = alertIssuer
+        self.pluginManager = pluginManager
+        self.staticSupportTypes = staticSupportTypes ??
+            (FeatureFlags.allowSimulators ? [MockSupport.self] : [])
+        staticSupportTypesByIdentifier = self.staticSupportTypes.reduce(into: [:]) { (map, type) in
+            map[type.supportIdentifier] = type
+        }
+
+        restoreState()
+
+        ((pluginManager?.availableSupports ?? [SupportUI]()) +
```

Comment:
This take a bit of time to grok. Consider breaking into three lines, one for calculating the static supports, one for collecting up all of the supports, and one for adding each of those. (Part of the problem is that there are three closures and it is hard to determine which is which.